src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -310,31 +309,24 @@ private static void populateLambdaArgs(Class<?> functionalInterface, final Class
             Method objectMethod = OBJECT_METHODS.get(m.getName());
             if (objectMethod != null && Arrays.equals(m.getTypeParameters(), objectMethod.getTypeParameters()))
               continue;
-
-            // Get functional interface's type params
-            Type returnTypeVar = m.getGenericReturnType();
```

Comment:
Whoops, turns out that I don't need to move this call after all. I'll fix this up while resolving the merge conflicts after the master build gets back into a stable state.